### DISCUSSION TITLE
RecursionError

### DISCUSSION THREAD

**Discussion by FerreroManuel** [py2exe/py2exe#165] [Q&A] [answered]
Hi, I always compress my console app with py2exe but now it's raising me a RecusionError. I've tried to compress the last version of my script, which was already compressed before without problems, but now it also raised the RecursionError.

This is part of the error (it's 3998 rows long):

> INFO:runtime:Analyzing the code
> Traceback (most recent call last):
>   File "C:\Dropbox\PC\Desktop\zprueba\modulos\freeze.py", line 29, in <module>
>     freeze(
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\__init__.py", line 190, in freeze
>     builder.analyze()
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\runtime.py", line 185, in analyze
>     target.analyze(mf)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\runtime.py", line 84, in analyze
>     modulefinder.run_script(self.script)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 201, in run_script
>     self.load_module('__main__', fp, pathname, stuff)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 409, in load_module
>     self.scan_code(co, m)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 482, in scan_code
>     self._safe_import_hook(name, m, fromlist, level=0)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 427, in _safe_import
>     self.import_hook(name, caller, level=level)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 215, in import_hook
>     q, tail = self.find_head_package(parent, name)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 271, in find_head_pa
>     q = self.import_module(head, qname, parent)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 368, in import_modul
>     m = self.load_module(fqname, fp, pathname, stuff)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 409, in load_module
>     self.scan_code(co, m)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 482, in scan_code
>     self._safe_import_hook(name, m, fromlist, level=0)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 427, in _safe_import
>     self.import_hook(name, caller, level=level)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 215, in import_hook
>     q, tail = self.find_head_package(parent, name)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 271, in find_head_pa
>     q = self.import_module(head, qname, parent)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 368, in import_modul
>     m = self.load_module(fqname, fp, pathname, stuff)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 409, in load_module
>     self.scan_code(co, m)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 482, in scan_code
>     self._safe_import_hook(name, m, fromlist, level=0)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 427, in _safe_import
>     self.import_hook(name, caller, level=level)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 215, in import_hook
>     q, tail = self.find_head_package(parent, name)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 271, in find_head_pa
>     q = self.import_module(head, qname, parent)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 368, in import_modul
>     m = self.load_module(fqname, fp, pathname, stuff)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 409, in load_module
>     self.scan_code(co, m)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 482, in scan_code
>     self._safe_import_hook(name, m, fromlist, level=0)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 427, in _safe_import
>     self.import_hook(name, caller, level=level)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 215, in import_hook
>     q, tail = self.find_head_package(parent, name)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 271, in find_head_pa
>     q = self.import_module(head, qname, parent)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 368, in import_modul
>     m = self.load_module(fqname, fp, pathname, stuff)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 409, in load_module
>     self.scan_code(co, m)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 482, in scan_code
>     self._safe_import_hook(name, m, fromlist, level=0)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 427, in _safe_import
>     self.import_hook(name, caller, level=level)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 215, in import_hook
>     q, tail = self.find_head_package(parent, name)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 271, in find_head_pa
>     q = self.import_module(head, qname, parent)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 368, in import_modul
>     m = self.load_module(fqname, fp, pathname, stuff)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 409, in load_module
>     self.scan_code(co, m)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 482, in scan_code
>     self._safe_import_hook(name, m, fromlist, level=0)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 427, in _safe_import
>     self.import_hook(name, caller, level=level)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 215, in import_hook
>     q, tail = self.find_head_package(parent, name)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 271, in find_head_pa
>     q = self.import_module(head, qname, parent)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 368, in import_modul
>     m = self.load_module(fqname, fp, pathname, stuff)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 381, in load_module
>     m = self.load_package(fqname, pathname)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 533, in load_package
>     self.load_module(fqname, fp, buf, stuff)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 409, in load_module
>     self.scan_code(co, m)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 505, in scan_code
>     self._safe_import_hook(name, m, fromlist, level=level)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 427, in _safe_import
>     self.import_hook(name, caller, level=level)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 215, in import_hook
>     q, tail = self.find_head_package(parent, name)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 271, in find_head_pa
>     q = self.import_module(head, qname, parent)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 368, in import_modul
>     m = self.load_module(fqname, fp, pathname, stuff)
>
> [...]
>
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 381, in load_module
>     m = self.load_package(fqname, pathname)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 533, in load_package
>     self.load_module(fqname, fp, buf, stuff)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 381, in load_module
>     m = self.load_package(fqname, pathname)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 533, in load_package
>     self.load_module(fqname, fp, buf, stuff)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 381, in load_module
>     m = self.load_package(fqname, pathname)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 533, in load_package
>     self.load_module(fqname, fp, buf, stuff)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 381, in load_module
>     m = self.load_package(fqname, pathname)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 533, in load_package
>     self.load_module(fqname, fp, buf, stuff)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 381, in load_module
>     m = self.load_package(fqname, pathname)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 533, in load_package
>     self.load_module(fqname, fp, buf, stuff)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 381, in load_module
>     m = self.load_package(fqname, pathname)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 533, in load_package
>     self.load_module(fqname, fp, buf, stuff)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 381, in load_module
>     m = self.load_package(fqname, pathname)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 533, in load_package
>     self.load_module(fqname, fp, buf, stuff)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 381, in load_module
>     m = self.load_package(fqname, pathname)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 533, in load_package
>     self.load_module(fqname, fp, buf, stuff)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 381, in load_module
>     m = self.load_package(fqname, pathname)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 533, in load_package
>     self.load_module(fqname, fp, buf, stuff)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 381, in load_module
>     m = self.load_package(fqname, pathname)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 533, in load_package
>     self.load_module(fqname, fp, buf, stuff)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 381, in load_module
>     m = self.load_package(fqname, pathname)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 533, in load_package
>     self.load_module(fqname, fp, buf, stuff)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 381, in load_module
>     m = self.load_package(fqname, pathname)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 533, in load_package
>     self.load_module(fqname, fp, buf, stuff)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 381, in load_module
>     m = self.load_package(fqname, pathname)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 533, in load_package
>     self.load_module(fqname, fp, buf, stuff)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 381, in load_module
>     m = self.load_package(fqname, pathname)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 533, in load_package
>     self.load_module(fqname, fp, buf, stuff)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 381, in load_module
>     m = self.load_package(fqname, pathname)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 533, in load_package
>     self.load_module(fqname, fp, buf, stuff)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 381, in load_module
>     m = self.load_package(fqname, pathname)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 533, in load_package
>     self.load_module(fqname, fp, buf, stuff)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 381, in load_module
>     m = self.load_package(fqname, pathname)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 533, in load_package
>     self.load_module(fqname, fp, buf, stuff)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 381, in load_module
>     m = self.load_package(fqname, pathname)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 533, in load_package
>     self.load_module(fqname, fp, buf, stuff)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 381, in load_module
>     m = self.load_package(fqname, pathname)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 533, in load_package
>     self.load_module(fqname, fp, buf, stuff)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 381, in load_module
>     m = self.load_package(fqname, pathname)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 533, in load_package
>     self.load_module(fqname, fp, buf, stuff)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 381, in load_module
>     m = self.load_package(fqname, pathname)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 533, in load_package
>     self.load_module(fqname, fp, buf, stuff)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 381, in load_module
>     m = self.load_package(fqname, pathname)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 533, in load_package
>     self.load_module(fqname, fp, buf, stuff)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 381, in load_module
>     m = self.load_package(fqname, pathname)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 531, in load_package
>     fp, buf, stuff = self.find_module("__init__", m.__path__)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 562, in find_module
>     return _find_module(name, path)
>   File "C:\Users\pety2\AppData\Roaming\Python\Python310\site-packages\py2exe\vendor\modulefinder.py", line 93, in _find_module
>     spec = importlib.machinery.PathFinder.find_spec(name, path)
>   File "<frozen importlib._bootstrap_external>", line 1430, in find_spec
>   File "<frozen importlib._bootstrap_external>", line 1402, in _get_spec
>   File "<frozen importlib._bootstrap_external>", line 1539, in find_spec
>   File "<frozen importlib._bootstrap_external>", line 1607, in _fill_cache
>   File "<frozen importlib._bootstrap_external>", line 1607, in <setcomp>
> RecursionError: maximum recursion depth exceeded while calling a Python object

Do you have any idea why it's happening? I tried updating py2exe, migrating from distutils.setup to py2exe.freeze and following all instructions from here, updating all python, packages... I don't know what else to do...

You can find the scripts that I need to freeze here: https://github.com/FerreroManuel/Morella-1/tree/master/modulos
They are ```caja.py```, ```cuentas.py```, ```mantenimiento.py```, ```rendiciones.py```, ```reporteador.py``` and ```ventas.py```.

Thank you so much for helping me.
Best regards. 
Manuel Ferrero.

**Comment by albertosottile** [comment, marked as answer]
Hi, this is not a discussion but rather an issue. I would encourage you to open an issue in [the issue tracker](https://github.com/py2exe/py2exe/issues). Please also provide a minimal example of the error, ideally with just one module (one .py file) to facilitate debugging.
